File: c/PE039.c
#include <stdio.h>

int main()
{
    int a, b, c, p, max = 0, size, maxp = 0;
    for(p = 12; p <= 1000; p++){
        size = 0;
        for(a = 3; a <= (p-3)/3; a++){
            for(b = a+1; b <= (p-4)/2; b++){
                c = p-a-b;
                if(c*c == a*a + b*b)
                    size++;
            }
        }
        if(size > max){
            max = size;
            maxp = p;
        }
    }
    printf("%d\n", maxp);
    return 0;
}

File: python/PE039.py
#!/usr/bin/env python

def numOfSolution(n):
    count = 0
    for a in range(3, n/3):
        for b in range(a+1, (n-a+1)/2):
            c = n - a - b
            if a*a + b*b == c*c:
                count += 1
    return count

def main():
    Max = 0
    for i in range(12, 1001):
        if numOfSolution(i) > Max:
            Max = numOfSolution(i)
            MaxP = i
    print MaxP

if __name__ == '__main__':
    main()
